### PR TITLE
Allow associated constants in `impl qobject::T`

### LINKED ISSUE #476: Allow associated constants in qobject::T
Especially when using roles in QAIMs, it's useful to associate a constant with each role, for more expressive code:
e.g.: 
``` rust
impl qobject::MarkdownCombiner {
    pub const NAME_ROLE : i32 = 0;
    pub const SELECTED_ROLE : i32 = 1;
}
```

However, cxx-qt currently does not allow this.

It does work on the native Rust struct (e.g. MarkdownCombiner). But that then requires more boilerplate to import the constants.

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
Closes #476

### CODE REVIEW

File: crates/cxx-qt-gen/src/generator/rust/qobject.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ use crate::{
     parser::qobject::ParsedQObject,
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/rust/qobject.rs:15]: could add a readme entry

Comment (LeonMatthesKDAB) [reply crates/cxx-qt-gen/src/generator/rust/qobject.rs:15]: Good point.

Comment (LeonMatthesKDAB) [reply crates/cxx-qt-gen/src/generator/rust/qobject.rs:15]: Done

File: crates/cxx-qt-gen/src/parser/qobject.rs [hunk 2]
```diff
@@ -205,10 +206,14 @@ impl ParsedQObject {
                 ImplItem::Method(method) => {
                     self.parse_impl_method(method)?;
                 }
+                ImplItem::Const(constant) => {
+                    self.passthrough_impl_items
+                        .push(ImplItem::Const(constant.clone()));
+                }
                 _ => {
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/parser/qobject.rs:213]: what about `ImplItem::Type`, `ImplType::Macro`, and `ImplType::Verbatim`, some or all of these seems relevant to pass through as well ? Eg there isn't much difference between Const and Type ?

File: crates/cxx-qt-gen/src/parser/qobject.rs [hunk 3]
```diff
@@ -205,10 +206,14 @@ impl ParsedQObject {
                 ImplItem::Method(method) => {
                     self.parse_impl_method(method)?;
                 }
+                ImplItem::Const(constant) => {
+                    self.passthrough_impl_items
+                        .push(ImplItem::Const(constant.clone()));
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/parser/qobject.rs:211]: assume you can't do `item.clone()` here ?

File: crates/cxx-qt-gen/src/parser/qobject.rs (hunk 2 above)
Comment (LeonMatthesKDAB) [reply crates/cxx-qt-gen/src/parser/qobject.rs:209]: Good point :+1: 
Now passed through as well

File: crates/cxx-qt-gen/src/parser/qobject.rs (hunk 3 above)
Comment (LeonMatthesKDAB) [reply crates/cxx-qt-gen/src/parser/qobject.rs]: I can, actually :D

**Review by ahayzen-kdab — APPROVED** [review]
LGTM

**Pull request merged**